packages/eslint-plugin/scripts/codegen.ts

```diff
@@ -0,0 +1,77 @@
+import type { Rule } from 'eslint'
+import fs from 'fs/promises'
+import { join, extname } from 'path'
+import { camelCase } from 'lodash'
+import { format, resolveConfig } from 'prettier'
+
+const header = (file: string) =>
+  `// THIS FILE WAS GENERATED DO NOT MODIFY BY HAND
+// @command yarn codegen:eslint
+` + file
+
+interface FoundRule {
+  module: Rule.RuleModule
+  moduleName: string
+}
+
+const ignore = ['index.ts']
+const rulesDir = join(__dirname, '../src/rules')
+const configsDir = join(__dirname, '../src/configs')
+
+async function generateConfig(name: string, rules: FoundRule[]) {
+  const code = `
+    export default {
+      plugins: ['@react-three'],
+      rules: {
+        ${rules.map((rule) => `'@react-three/${rule.moduleName}': 'error'`)}
+      },
+    }
+  `
+
+  const filepath = join(configsDir, `${name}.ts`)
+  await writeFile(filepath, code)
+}
+
+async function writeFile(filepath: string, code: string) {
+  const config = await resolveConfig(filepath)
+  await fs.writeFile(filepath, format(header(code), { ...config, filepath }))
+}
+
+async function generateRuleIndex(rules: FoundRule[]) {
+  const code = `
+    ${rules.map((rule) => `import ${camelCase(rule.moduleName)} from './${rule.moduleName}'`)}
+
+    export default {
+    ${rules.map((rule) => `'${rule.moduleName}': ${camelCase(rule.moduleName)}`)}
+    }
+  `
+
+  const filepath = join(rulesDir, 'index.ts')
+  await writeFile(filepath, code)
+}
+
+async function generate() {
```

Comment:
Surprised this doesn't exist as a standalone module! Let me know if it does and my searchfu needs work.